Src/FluentAssertions/Equivalency/Matching/MustMatchByNameRule.cs

```diff
@@ -36,6 +36,13 @@ public IMember Match(IMember expectedMember, object subject, INode parent, IEqui
                     $"Expectation has {expectedMember.Description} that the other object does not have.");
             }
 
+            if (config.IgnoreNonBrowsableOnSubject && !subjectMember.IsBrowsable)
```

Comment:
Some observations:

I'm not sure what case we are testing here (I've been a bit out of the loop) and no tests fails if I remove it.

As this is defined in `MustMatchByNameRule` it won't be executed when combing with `ExcludingMissingMembers`.

This passes
```cs
var subject = new NonBrowable { Browsable = 1, NonBrowsable = 2 };
var expected = new NonBrowable { Browsable = 1, NonBrowsable = 3 };

subject.Should().BeEquivalentTo(expected, opt => opt
    .IgnoringNonBrowsableMembersOnSubject());
```

This fails
```cs
var subject = new NonBrowable { Browsable = 1, NonBrowsable = 2 };
var expected = new NonBrowable { Browsable = 1, NonBrowsable = 3 };

subject.Should().BeEquivalentTo(expected, opt => opt
    .ExcludingMissingMembers()
    .IgnoringNonBrowsableMembersOnSubject());
```

Reply:
Mmm, indeed, this isn't covered by a test. This was added in response to feedback from @dennisdoomen. You can see it in action by removing the call to `.ExcludingMissingMembers` from the test `When_property_is_non_browsable_only_in_subject_ignoring_non_browsable_members_on_subject_should_make_it_succeed`.

Reply:
I've added a test that covers this case in f730ff4. It fails if the lines in question are removed from `MustMatchByNameRule.cs`.

Reply:
The short explanation is that `.IgnoringNonBrowsableMembersOnSubject()` is supposed to pretend that the members _don't exist on the subject at all_, which means that if they _are_ present on the expectation, then without `.ExcludingMissingMembers()`, it should fail the equivalency test -- but without these lines, that failure doesn't happen because `MustMatchByNameRule` has already finished executing before `IgnoreNonBrowsableOnSubject` is processed by `FindMatchFor` in `StructuralEqualityEquivalencyStep`, and the member isn't _actually_ physically missing.